.github/workflows/ci.yml

```diff
@@ -0,0 +1,29 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
+
+permissions:
+  contents: read
+
+jobs:
+  build-test:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.25.4'
+          cache: true
+
+      - name: Run CI checks
+        run: bash ./ci-check.sh
```

Comment:
Hopefully this is right. Not sure what you'll have to do. Intent is to run on all branches - not allow breaking changes in.